WordPressShared/WordPressShared/Core/Utility/RichContentFormatter.swift

```diff
@@ -106,42 +114,147 @@ import Foundation
                                                                     options: .reportCompletion,
-                                                                    range: NSRange(location: 0, length: content.characters.count),
+                                                                    range: NSRange(location: 0, length: content.count),
                                                                     withTemplate: closePTag)
 
         // Remove duplicate/redundant p tags.
         content = RegEx.pTagsStart.stringByReplacingMatches(in: content,
                                                                    options: .reportCompletion,
-                                                                   range: NSRange(location: 0, length: content.characters.count),
+                                                                   range: NSRange(location: 0, length: content.count),
                                                                    withTemplate: openPTag)
 
         content = RegEx.pTagsEnd.stringByReplacingMatches(in: content,
                                                                    options: .reportCompletion,
-                                                                   range: NSRange(location: 0, length: content.characters.count),
+                                                                   range: NSRange(location: 0, length: content.count),
                                                                    withTemplate: closePTag)
 
         content = filterNewLines(content)
 
         return content
     }
 
-
+    
+    /// Converts DIV with class gallery class to gallery tags
+    ///
+    /// - Parameters:
+    ///     - string: The content string to format.
+    ///
+    /// - Returns: The formatted string.
+    ///
+    public class func normalizeGallery(_ string : String) -> String {
+        
+        guard string.count > 0 else {
+            return string
+        }
+        
+        
+        //Types of WP Galleries
+        let iconGallery = "class='gallery"
+        let tiledGallery = "class=\"tiled-gallery"
+        
+        //Gallery Tag for identification
+        let galleryStart = "<gallery>"
+        let galleryEnd = "</gallery>"
+        
+        var content = string
+        
+        //Scan and Insert Indentifier for Gallery Start and End
+        content = scannerForGalleryBy(galleryType: iconGallery, inString: content)
+        content = scannerForGalleryBy(galleryType: tiledGallery, inString: content)
+        
+        
+        //Replace Div with Gallery tag
+        content = RegEx.galleryDivStart.stringByReplacingMatches(in: content, options: .reportCompletion, range: NSRange(location: 0, length: content.count), withTemplate: galleryStart)
+        content = RegEx.galleryDivEnd.stringByReplacingMatches(in: content, options: .reportCompletion, range: NSRange(location: 0, length: content.count), withTemplate: galleryEnd)
```

Comment:
My thinking behind this was to the keep the logic a bit more simple.  This function `scanForGalleryBy(galleryType : String, inString : String)` only scans and identifies where a gallery should start and end.  Then clean it up with the RegEx, rather than using another Scanner.  It could be done in one go, but this felt like a safe, reliable option to me.